### ISSUE TITLE
[WASM] Understanding 'BindableAttribute'

### ISSUE THREAD

**Issue by DierkDroth** [unoplatform/uno#6267]
I'm trying to get my head around this warning
`The Bindable attribute is missing and the type [XXXX] is not known by the MetadataProvider. Reflection was used instead of the binding engine and generated static metadata. Add the Bindable attribute to prevent this message and performance issues.`

I'm aware of [this ](https://github.com/unoplatform/uno/issues/1688) but still have some questions:
- we found that adding either System.ComponentModel.BindableAttribute(BindableSupport.Default) (is this the correct parameter used?!?) or Windows.UI.Xaml.Data.BindableAttribute eliminates that warning (note, the former is needed for assemblies which don't have any dependency on UWP/WinUI). Is that expected?
- I'm not clear on the purpose of this attribute nor that warning at all. Why would I have to mark a class with that attribute and I why couldn't UNO just assume a class would be 'bindable' by default? What prevents a class from being 'bindable'? And shouldn't the logic rather be the other way round like 'mark as non-bindable', since this appears to be the way rarer case, no?!?
- I'm getting warning above on [Uno.UI.Xaml.Controls.SystemFocusVisual] which I don't control. How could I get rid of that warning?

I would appreciate if some light could be shed. Thanks in advance

**Comment by jeromelaban** [comment]
Thanks for the report. 

I've added [this documentation](https://github.com/unoplatform/uno/pull/6271/files) to shed more light on the process.

With regards to the second point, classes can be used by the binding engine, but to improve the performance adding the `Bindable` attribute is needed. As we generate code to improve performance, we can't generate code arbitrarily for all types found in the app (it would be a problem for the app size).

Finally, the last point is a fix that needs to be made inside Uno.

**Comment by DierkDroth** [comment]
@jeromelaban while ripping through my code I found that these classes throw the same warning
- [Windows.UI.Xaml.Controls.VirtualizingPanelLayout]
- [Windows.UI.Xaml.CornerRadius]

Could the warning be eliminated there as well or is this beyond UNO's control?

**Comment by DierkDroth** [comment]
Great, thanks

**Comment by marwalsch** [comment]
@jeromelaban I am witnessing the same messages on iOS for the following classes when targeting WinUI

```
[Microsoft.UI.Xaml.Controls.FlipView]
[Microsoft.UI.Xaml.Shapes.Shape]
[Microsoft.UI.Xaml.Controls.ProgressBar]
[Microsoft.UI.Xaml.Controls.ProgressBarTemplateSettings]
[Microsoft.UI.Xaml.Controls.SwipeControl]
[Microsoft.UI.Xaml.Controls.ListViewItem]
[Microsoft.UI.Xaml.Controls.Primitives.ListViewItemTemplateSettings]
```

as well as custom `UserControl`s decorated with `System.ComponentModel.BindableAttribute`. Is this expected or should this have been fixed by [#6274](https://github.com/unoplatform/uno/pull/6274) as well?

**Comment by jeromelaban** [comment]
@marwalsch what you're seeing is an uno 4.0 specific issue, and is fixed in Uno 4.1-dev builds.
